library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -0,0 +1,198 @@
+    VALID_LINE_REGEX = /\A\w+\s+\| \d+/
+
+    attr_reader :number, :snapshot_type, :previous_number, :timestamp, :user,
+      :cleanup_algo, :description
+
+    # Determines whether snapper is configured or not
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), FIND_CONFIG_CMD)
+      log.info("Checking if Snapper is configured: \"#{FIND_CONFIG_CMD}\" returned: #{out}")
+      out["exit"] == 0
+    end
+
+    # Creates a new 'single' snapshot
+    #
+    # @param description [String] Snapshot's description.
+    # @return [FsSnapshot] The created snapshot.
+    #
+    # @see FsSnapshot.create
+    def self.create_single(description)
+      create(:single, description)
+    end
+
+    # Creates a new 'pre' snapshot
+    #
+    # @param description [String] Snapshot's description.
+    # @return [FsSnapshot] The created snapshot.
+    #
+    # @see FsSnapshot.create
+    def self.create_pre(description)
+      create(:pre, description)
+    end
+
+    # Creates a new 'post' snapshot
+    #
+    # Each 'post' snapshot corresponds with a 'pre' one. If the number of the
+    # 'pre' snapshot is not given, it will try to figure out that number.
+    #
+    # @param description     [String] Snapshot's description.
+    # @param previous_number [Fixnum] Number of the previous snapshot
+    # @return [FsSnapshot] The created snapshot.
+    #
+    # @see FsSnapshot.create
+    def self.create_post(description, previous_number = nil)
+      previous =
+        if previous_number.nil?
+          last = all.reverse.find { |s| s.snapshot_type == :pre || s.snapshot_type == :post }
+          (last && last.snapshot_type == :pre) ? last : nil
+        else
+          find(previous_number)
+        end
+
+      if previous
+        create(:post, description, previous)
+      else
+        log.error "Previous filesystem snapshot was not found"
+        raise PreviousSnapshotNotFound
+      end
+    end
+
+    # Creates a new snapshot
+    #
+    # It raises and exception if Snapper is not configured or if snapshot
+    # creation fails.
+    #
+    # @param snapshot_type [Symbol]    Snapshot's type: :pre, :post or :single.
+    # @param description   [String]    Snapshot's description.
+    # @param previous      [FsSnashot] Previous snapshot.
+    # @return [FsSnapshot] The created snapshot if the operation was
+    #                      successful.
+    private_class_method def self.create(snapshot_type, description, previous = nil)
+      raise SnapperNotConfigured unless configured?
+
+      cmd = CREATE_SNAPSHOT_CMD % [snapshot_type, description]
+      cmd << " --pre-num %s" % previous.number unless previous.nil?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), cmd)
+      if out["exit"] == 0
+        find(out["stdout"].to_i) # The CREATE_SNAPSHOT_CMD returns the number of the new snapshot.
```

Comment:
I think it make sense to log content of stdout, so we know number of created snapshot, because there is a risk if output slightly change, that `to_i` return `0` which can be quite hard to debug.